legal-api/src/legal_api/models/amalgamation.py

```diff
@@ -82,33 +79,26 @@ def json(self):
         }
 
     @classmethod
-    def get_amalgamation_revision_obj(cls, transaction_id, business_id):
+    def get_amalgamation_revision_obj(cls, filing, legal_entity_id):
         """Get amalgamation for the given transaction id."""
         # pylint: disable=singleton-comparison;
-        amalgamation_version = version_class(Amalgamation)
+        amalgamation_version = history_cls(Amalgamation)
```

Comment:
I'm stuck on fixing one last test and I am not sure why it is failing. The test failing is:

`tests/unit/models/test_legal_entity.py::test_amalgamated_into_business_json[EXIST-2] - AttributeError: 'NoneType' object has no attribute 'amalgamation_date'`

I can trace it back to this function, `get_amalgamation_revision_obj`, which returns `None` from the below query. From what I can tell, the `amalgamations_history` table does not get populated during this test (checking the count confirms this), so the query returns None.

Does anyone know how the history table works, and why it isn't updating for the `test_amalgamated_into_business_json` unit test? I'm sure I am missing something in unit test itself.

Reply:
I think it's because the amalgamation record would not be in the `amalgamation_history` table in your unit test scenario  

The new versioning works in a way where the latest changes only reside in the parent table which is not how the old versioning worked.  The old versioning model always cut a row for latest change even for the versioning table.

In  the failing test scenario, there will only one entry the `amalgamation` table with no row in `amalgamation_history`.

An update update like this should work I think
![image](https://github.com/bcgov/lear/assets/6685223/0703ef13-05c4-4cfe-8b0e-108d36f38fe4)

Reply:
Ahh I see, that makes sense. That works perfectly thanks for the help Argus!